src/testMPAppIDs.cpp

```diff
@@ -0,0 +1,44 @@
+#ifndef TESTMPAPPIDS_H
+#define TESTMPAPPIDS_H
+
+#include "pmpo_createTestMPMesh.hpp"
+#include "pmpo_defines.h"
+#include "pmpo_c.h"
+
+extern "C" {
+  void testAppIDPointer(MPMesh_ptr p_mpmesh, const int numMPs, const int* allMP2Elm);
+}
+
+void testAppIDPointer(MPMesh_ptr p_mpmesh, 
+                              const int numMPs, // total number of MPs which is GREATER than or equal to number of active MPs
+                              const int* allMP2Elm) {
+  auto p_MPs = ((polyMPO::MPMesh*)p_mpmesh)->p_MPs;
+  PMT_ALWAYS_ASSERT(numMPs >= p_MPs->getCount());
+  PMT_ALWAYS_ASSERT(numMPs >= p_MPs->getMaxAppID());
+
+  int internalMPCapacity = p_MPs->getCapacity(); // pumipic expects full capacity to rebuild
+  auto mp2Elm = create_mirror_view_and_copy(allMP2Elm, internalMPCapacity);
+  Kokkos::View<int*> mp2Elm_d("mp2Elm_d", internalMPCapacity);
+  Kokkos::deep_copy(mp2Elm_d, mp2Elm);
+
+  int numAddedMPs = 2;
+  Kokkos::View<int*> added_mp2Elm_d("added_mp2Elm_d", numAddedMPs);
+  Kokkos::parallel_for("set addedMP2Elm", numAddedMPs, KOKKOS_LAMBDA (const int i) {
+    added_mp2Elm_d(i) = i;
+  });
+
+  std::vector<int> added_mpIDs(numAddedMPs);
+  for(int i=0; i<numAddedMPs; i++)
+    added_mpIDs[i] = p_MPs->getNextAppID();
+  auto added_mpIDs_d = create_mirror_view_and_copy(added_mpIDs.data(), numAddedMPs);
+
+  p_MPs->rebuild(mp2Elm_d, added_mp2Elm_d, added_mpIDs_d);
```

Comment:
Does pumipic allow sending empty or null pointer for 'mp2Elm_d' (first argument)? What if there is no change in any current MP. If pumipic doesn't allow that, then we need to add function 'p_MPs->rebuild(added_mp2Elm_d, added_mpIDs_d);' and inside this function pass mp2Elm for current/existing MPs on device.

Reply:
I does not. I created the rebuild function in polyMPO